docs/src/components/Sidebar/Sidebar.tsx

```diff
@@ -63,69 +64,23 @@ class Sidebar extends React.Component<any, any> {
     const hasModifier = e.altKey || e.ctrlKey || e.metaKey
     const bodyHasFocus = document.activeElement === document.body
 
-    if (!hasModifier && isAZ && bodyHasFocus) this._searchInput.focus()
+    if (!hasModifier && isAZ && bodyHasFocus) this.searchInputRef.current.focus()
   }
 
-  handleItemClick = () => {
+  handleItemClick = e => {
     const { query } = this.state
 
-    if (query) this.setState({ query: '' })
-    // TODO: as part of search input re-enabling
-    // if (document.activeElement === this._searchInput) this._searchInput.blur()
+    if (query) {
+      this.setState({ query: '' })
+      const at = e.target.href.replace(e.target.baseURI, '/')
+      const categoryIndex = this.findActiveCategoryIndex(at, this.getSectionsWithoutSearchFilter())
+      this.setState({ activeCategoryIndex: categoryIndex })
+    }
   }
 
-  /* TODO: as part of search input re-enabling
-    private renderSearchItems = () => {
-      const { selectedItemIndex, query } = this.state
-      if (!query) return undefined
-
-      let itemIndex = -1
-      const startsWithMatches: ComponentMenuItem[] = []
-      const containsMatches: ComponentMenuItem[] = []
-      const escapedQuery = _.escapeRegExp(query)
-
-      _.each(componentMenu, info => {
-        if (new RegExp(`^${escapedQuery}`, 'i').test(info.displayName)) {
-          startsWithMatches.push(info)
-        } else if (new RegExp(escapedQuery, 'i').test(info.displayName)) {
-          containsMatches.push(info)
-        }
-      })
-
-      this.filteredMenu = [...startsWithMatches, ...containsMatches]
-      const menuItems = _.map(this.filteredMenu, info => {
-        itemIndex += 1
-        const isSelected = itemIndex === selectedItemIndex
-
-        if (isSelected) this.selectedRoute = getComponentPathname(info)
-
-        return (
-          <Menu.Item
-            key={info.displayName}
-            content={info.displayName}
-            onClick={this.handleItemClick}
-            active={isSelected}
-            as={NavLink}
-            to={getComponentPathname(info)}
-          />
-        )
-      }, this.filteredMenu)
-
-      return (
-        <Menu.Item
-          key={info.displayName}
-          name={info.displayName}
-          onClick={this.handleItemClick}
-          active={isSelected}
-          as={NavLink}
-          to={getComponentPathname(info)}
-        >
-          {info.displayName}
-          {isSelected && selectedItemLabel}
-        </Menu.Item>
-      )
-    })
-*/
+  treeActiveIndexChanged = data => {
```

Comment:
Usually these kind of callbacks sends two params here:
- event
- all props

So this one should be refactored to something like:
```
treeActiveIndexChanged = (e, props) => {
 this.setState({activeCategoryIndex: props.activeIndex})
}
```

Reply:
Take a look on the `onMenuOpenChange` in the `MenuItem` component.